modules/core/src/v2/wallet.ts

```diff
@@ -1168,8 +1171,13 @@ Wallet.prototype.prebuildAndSignTransaction = function(params, callback) {
     // call prebuildTransaction and keychains-get in parallel
     // the prebuild can be overridden by providing an explicit tx
     const txPrebuildQuery = params.prebuildTx ? Promise.resolve(params.prebuildTx) : this.prebuildTransaction(params);
-    const userKeychainQuery = this.baseCoin.keychains().get({ id: this._wallet.keys[0], reqId: params.reqId });
-    const [txPrebuild, userKeychain] = yield Promise.all([txPrebuildQuery, userKeychainQuery]);
+
+    const self = this;
+    const keychains = yield Promise.all(function keychainQueriesForSigning() {
```

Comment:
Let's pull this out above the call to promise.all. I don't think the IIFE is needed here

Reply:
ok